modules/caddyhttp/matchers.go

```diff
@@ -909,25 +923,47 @@ func (m MatchRemoteIP) getClientIP(r *http.Request) (net.IP, error) {
 	if err != nil {
 		ipStr = remote // OK; probably didn't have a port
 	}
+	// Some IPv6-Adresses can contain zone identifiers at the end,
+	// which are separated with "%"
+	if strings.Contains(ipStr, "%") {
+		split := strings.Split(ipStr, "%")
+		ipStr = split[0]
+		zoneID = split[1]
+	}
 	ip := net.ParseIP(ipStr)
 	if ip == nil {
-		return nil, fmt.Errorf("invalid client IP address: %s", ipStr)
+		return nil, zoneID, fmt.Errorf("invalid client IP address: %s", ipStr)
 	}
-	return ip, nil
+	return ip, zoneID, nil
 }
 
 // Match returns true if r matches m.
 func (m MatchRemoteIP) Match(r *http.Request) bool {
-	clientIP, err := m.getClientIP(r)
+	clientIP, zoneID, err := m.getClientIP(r)
 	if err != nil {
 		m.logger.Error("getting client IP", zap.Error(err))
 		return false
 	}
-	for _, ipRange := range m.cidrs {
+	zoneFilter := true
+	for i, ipRange := range m.cidrs {
 		if ipRange.Contains(clientIP) {
-			return true
+			// Check if the filter ips contain zone identifiers.
+			if m.zones[i] == "" {
+				return true
+			}
+			// Check if the incomming zoneID is that one in filter.
+			// Assign true or false for the handling of multible ips with or without zoneIDs in Filter
+			// and handling of the "zoneID not match" debug info.
+			if zoneID == m.zones[i] {
+				zoneFilter = true
+				return true
+			}
```

Comment:
It has to, because it could be that there is an IP with zone before, that already set the zoneFilter to false. If now an other IP with the right zone matches, is has to set the filter to true. Otherwise the debug info later would be given out wrongly.

Reply:
But `return true` means that it won't continue at all, it'll exit the function right then.

Reply:
Also with this change, the above condition and this one could be merged to just `if m.zones[i] == "" || zoneID == m.zones[i] { return true`

Reply:
> But `return true` means that it won't continue at all, it'll exit the function right then.

The ``for`` loop runs for every ip in every ``remote_ip`` filter. Only serveral different remote_ip filters (with one or more ips and ranges) are separated from each other. Without the setting of ``true`` it will stay ``wrong``, even if thas isn't right.

Reply:
To ne honest, im sure it's a very rare usecase and it's also not terrible, if you get a wrong debug message, but yeah, it's not correct...

Reply:
Okay. You are right! ;D